FF1Lib/ExtensiveHints/HintSources/FloorHintsHintSource.cs

```diff
@@ -1,108 +1,108 @@
-﻿using RomUtilities;
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
-using static FF1Lib.FF1Rom;
-
-namespace FF1Lib
-{
-	public class FloorHintsHintSource : BaseHintSource
-	{
-		ItemPrices Prices;
-
-		public FloorHintsHintSource(MT19337 _rng, NPCdata _npcData, Flags _flags, OverworldMap _overworldMap, FF1Rom _rom) : base(_rng, _npcData, _flags, _overworldMap, _rom)
-		{
-			Prices = new ItemPrices(_rom);
-		}
-
-		public override List<GeneratedHint> GetHints()
-		{
-			List<GeneratedHint> hints = GenerateFloorHints();
-
-			return hints;
-		}
-
-		private List<GeneratedHint> GenerateFloorHints()
-		{
-			List<GeneratedHint> hints = new List<GeneratedHint>();
-
-			HashSet<Item> WowItems = new HashSet<Item>(ItemLists.UberTier.Concat(ItemLists.LegendaryWeaponTier).Concat(ItemLists.LegendaryArmorTier).Concat(ItemLists.AllQuestItems).Distinct());
-			HashSet<Item> MehItems = new HashSet<Item>(ItemLists.RareWeaponTier.Concat(ItemLists.RareArmorTier).Distinct());
-			HashSet<Item> SparkleItems = new HashSet<Item>(ItemLists.AllGoldTreasure);
-
-			var locations = rom.generatedPlacement.Where(p => p is TreasureChest).GroupBy(p => p.MapLocation);
```

Comment:
Why is var not allowed here?

Reply:
because the current rules chooise IDE0008 over IDE0007

https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0007-ide0008

Personally, I do prefer explicit types, but this is all up for discussion

Reply:
I like var (I mean, it's less work), but since we're manipulating a lot of lists and arrays explicit types are way clearer at first glance and should make it easier to newcomers to figure out what the randomizer is doing

Reply:
one thing that is nice, is you can program it with var, then run the formatter and it will try to replace it with the correct type (it only failed a couple times for me and that was when I had a config issue resulting in the formatter not understanding what a Blob is)

Reply:
2 years ago i preferred explicit types as well. But when using linq, things can get out of hands very easily like:

`foreach (IEnumerable<IGrouping<KeyValuePair<string, Tuple<string, string>>> item in list)`

and become really readable.

Reply:
Hmm.  I like `var` for pretty much everything except primitive types like `int` and `ushort`.  Usually when you're doing a complicated LINQ statement, you don't really care what the exact type is because you're just going to iterate over it in a sec.  You can always hover to see the type anyway if you need to.